### PR TITLE
Add options `nix.nixPath` `nix.registry`

### PR DESCRIPTION
**Pull request by azuwis**
Also rename `nix.extraConfig` to `nix.extraOptions`

Most of the codes are copied from NixOS modules.

### CODE REVIEW

File: modules/environment/nix.nix [hunk 1]
```diff
@@ -63,28 +73,32 @@ in
...
-      '';
-    };
+  config = mkMerge [
+    {
+      environment.etc = {
+        "nix/nix.conf".text = ''
+          sandbox = false
+          substituters = ${concatStringsSep " " cfg.substituters}
+          trusted-public-keys = ${concatStringsSep " " cfg.trustedPublicKeys}
+          ${cfg.extraConfig}
```
Comment (t184256) [inline modules/environment/nix.nix:142]: needs to be renamed, or there will be warnings by default

**Comment by t184256** [comment]
Looks OK to me.

1. one renaming needed, comment inline
2. please bump the year
3. ideally mimic style of other commit messages
4. it looks largely lifted from nixpkgs, please add attribution. smth like

```
# Based on
# https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/nix-daemon.nix
# (Copyright (c) 2003-2022 Eelco Dolstra and the Nixpkgs/NixOS contributors,
#  licensed under MIT License as well)
```

**Review by t184256 — CHANGES_REQUESTED** [review]

**Comment by azuwis** [comment]
Please review the new changes.

**Review by t184256 — APPROVED** [review]
Thank you!

**Pull request merged**